test/powershell/engine/Basic/Encoding.Tests.ps1

```diff
@@ -0,0 +1,74 @@
+Describe "File encoding tests" -tag CI {
+
+    Context "ParameterType for parameter 'Encoding' should be 'Encoding'" {
+        BeforeAll {
+            $testCases = get-command -module Microsoft.PowerShell.* | 
+                Where-Object { $_.Parameters -and $_.Parameters['Encoding'] } |
+                ForEach-Object { @{ Command = $_ } }
+        }
+        It "Encoding parameter of command '<Command>' is type 'Encoding'" -testcase $testCases {
+            param ( $command )
+            $command.Parameters['Encoding'].ParameterType.FullName | Should Be "System.Text.Encoding"
+        }
+    }
+    Context "File contents are UTF8 without BOM" {
+        BeforeAll {
+            $testStr = "t" + ([char]233) + "st"
+            $nl = [environment]::newline
+            $utf8Bytes = 116,195,169,115,116 
+            $nlBytes = [byte[]][char[]]$nl
+            $ExpectedWithNewline = $( $utf8Bytes ; $nlBytes )
+            $outputFile = "${TESTDRIVE}/file.txt"
+            $utf8Preamble = [text.encoding]::UTF8.GetPreamble()
+            $simpleTestCases = @(
+                # new-item does not add CR/NL
+                @{ Command = "new-item"; Parameters = @{ type = "file";value = $testStr; Path = $outputFile }; Expected = $utf8Bytes ; Operator = "be" }
+                # the following commands add a CR/NL
+                @{ Command = "set-content"; Parameters = @{ value = $testStr; Path = $outputFile }; Expected = $ExpectedWithNewline ; Operator = "be" }
+                @{ Command = "add-content"; Parameters = @{ value = $testStr; Path = $outputFile }; Expected = $ExpectedWithNewline ; Operator = "be" }
+                @{ Command = "out-file"; Parameters = @{ InputObject = $testStr; Path = $outputFile }; Expected = $ExpectedWithNewline ; Operator = "be" }
+                # Redirection
+                @{ Command = { $testStr > $outputFile } ; Expected = $ExpectedWithNewline ; Operator = "be" }
+                )
+            function Get-FileBytes ( $path ) {
+                [io.file]::ReadAllBytes($path)
+            }
+        }
+
+        AfterEach { 
+            if ( test-path $outputFile ) {
+                remove-item -force $outputFile
+            }
+        }
+
+        It "<command> produces correct content '<Expected>'" -testcases $simpleTestCases {
+            param ( $Command, $parameters, $Expected, $Operator)
+            & $command @parameters
+            $bytes = Get-FileBytes $outputFile
+            $bytes -join "-" | should ${Operator} ($Expected -join "-")
```

Comment:
If the operator is always `Be` we can directly use it here and remove it from `$simpleTestCases`

Reply:
Should this be `BeExactly`?